src/firefly/js/ui/tap/TapUtil.js

```diff
@@ -218,13 +218,24 @@ export function getColumnAttribute(columnsModel, colName, attrName) {
 
 const hasElements= (a) => Boolean(isArray(a) && a?.length);
 
+function mergeAdditionalServices(tapServices, additional) {
+    if (!hasElements(additional)) return tapServices;
+
+    const modifiedOriginal= tapServices.map( (t) => {
+        const match= additional.find( (a) => a.label===t.label);
+        return match ? {...t,...match} : t;
+    });
+    const trulyAdditional= additional.filter( (a) => tapServices.filter( (t) => t.label===a.label).length===0 );
+
```

Comment:
Would a `tapServices.find()` be more appropriate here?

Reply:
yea, I should use `find`